app/uk/gov/hmrc/eoricommoncomponent/frontend/services/ApplicationService.scala

```diff
@@ -35,65 +40,60 @@ class ApplicationService @Inject() (
 )(implicit ec: ExecutionContext)
     extends EnrolmentExtractor {
 
-  private def isEnrolmentInUse(service: Service, groupId: String, loggedInUser: LoggedInUserWithEnrolments)(implicit
-    hc: HeaderCarrier
-  ): Future[Option[ExistingEori]] =
-    groupEnrolment.groupIdEnrolments(groupId).flatMap { groupEnrolments =>
-      existingEoriForUserOrGroup(loggedInUser, groupEnrolments) match {
-        case Some(existingEori) => enrolmentStoreProxyService.isEnrolmentInUse(service, existingEori)
-        case _                  => Future.successful(None)
-      }
+  private def getEnrolmentInUse(
+    service: Service,
+    groupEnrolments: List[EnrolmentResponse],
+    loggedInUser: LoggedInUserWithEnrolments
+  )(implicit hc: HeaderCarrier): Future[Option[ExistingEori]] =
+    existingEoriForUserOrGroup(loggedInUser, groupEnrolments) match {
+      case Some(existingEori) => enrolmentStoreProxyService.isEnrolmentInUse(service, existingEori)
+      case _                  => Future.successful(None)
     }
 
-  private def isUserEnrolledFor(loggedInUser: LoggedInUserWithEnrolments, service: Service): Boolean =
-    enrolledForService(loggedInUser, service).isDefined
-
-  private def isUserEnrolledForOtherServices(loggedInUser: LoggedInUserWithEnrolments): Boolean =
+  private def isUserEnrolledToOtherService(loggedInUser: LoggedInUserWithEnrolments) =
     enrolledForOtherServices(loggedInUser).isDefined
 
-  private def checkAllServiceEnrolments(loggedInUser: LoggedInUserWithEnrolments, groupId: String)(implicit
-    hc: HeaderCarrier,
-    request: Request[_]
-  ): Future[Either[JourneyError, JourneyType]] =
-    if (isUserEnrolledForOtherServices(loggedInUser))
-      Future.successful(Right(AutoEnrolment))
-    else
-      groupEnrolment.checkAllServiceEnrolments(groupId).flatMap {
-        case Some(groupEnrolment) if groupEnrolment.eori.isDefined =>
-          cache.saveGroupEnrolment(groupEnrolment).map(_ => Right(AutoEnrolment))
-        case _ =>
-          Future.successful(Right(LongJourney))
-      }
+  private def isGroupEnrolledToOtherService(groupEnrolments: List[EnrolmentResponse])(implicit request: Request[_]) = {
+    def checkSupportedServiceEnrolments(groupEnrolments: List[EnrolmentResponse]) = {
+      val serviceList = Service.supportedServicesMap.values.toList
+      groupEnrolments.find(enrolment => serviceList.contains(enrolment.service))
+    }
 
-  private def checkCDSEnrolments(loggedInUser: LoggedInUserWithEnrolments, groupId: String)(implicit
-    hc: HeaderCarrier,
-    request: Request[_]
-  ): Future[Either[JourneyError, JourneyType]] =
-    if (isUserEnrolledFor(loggedInUser, Service.cds))
-      Future.successful(Right(AutoEnrolment))
+    checkSupportedServiceEnrolments(groupEnrolments) match {
+      case Some(groupEnrolment) if groupEnrolment.eori.isDefined =>
+        cache.saveGroupEnrolment(groupEnrolment).map(_ => true)
```

Comment:
I appreciate that we're retaining this because that's what was happening before, but hidden calls to the cache with methods that are 'IsXYZTrue' is something we should be avoiding.

In this case I do think we might have to look at going beyond simple boolean returns